lib/schemas.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable new-cap */
+'use strict';
+
+const {
+  DATA_UNIT_TYPE,
+
+  HANDSHAKE,
+  PARCEL_HEADER,
+  STREAM_HEADER,
+  CHUNK_HEADER,
+  CHUNK_PAYLOAD,
+  ACK,
+
+  DATA_UNIT_TYPES,
+  ENCODINGS,
+  COMPRESSIONS,
+  HANDSHAKE_STATUSES,
+} = require('./constants');
+
+const UInt = {
+  parse: (buffer, size, offset = 0) => buffer.readUIntLE(offset, size),
+};
+
+const Binary = {
+  parse: (buffer, size, offset = 0) => buffer.slice(offset, offset + size),
+};
+
+const Enum = values => ({
+  parse: (buffer, size, offset = 0) => values[buffer.readUIntLE(offset, size)],
+});
+
+const Struct = schemas => ({
+  parse: (buffer, size, offset = 0) => {
+    const obj = {};
+    for (const { name, size, type } of schemas) {
+      obj[name] = type.parse(buffer, size, offset);
+      offset += size;
+    }
+    return obj;
+  },
+});
+
+module.exports = [
+  {
+    name: DATA_UNIT_TYPE,
+    size: 1,
+    type: Enum(DATA_UNIT_TYPES),
+  },
+  {
+    name: HANDSHAKE,
+    size: 35,
+    type: Struct([
+      { name: 'version', size: 2, type: UInt },
+      { name: 'status', size: 1, type: Enum(HANDSHAKE_STATUSES) },
+      { name: 'token', size: 32, type: Binary },
+    ]),
+  },
+  {
+    name: PARCEL_HEADER,
+    size: 7,
+    type: Struct([
+      { name: 'id', size: 4, type: UInt },
+      { name: 'size', size: 2, type: UInt },
+      { name: 'compression', size: 1, type: Enum(COMPRESSIONS) },
```

Comment:
It may be useful to create a type `const Compression = Enum(COMPRESSIONS);` so later it would be possible to check the type of the schema like `schema.type === Compression`.

Reply:
As discussed online, we won't do it this because of YAGNI